test/test_sparse_csr.py

```diff
@@ -2846,7 +2846,7 @@ def _to_from_layout(layout_a, layout_b, a):
                 # change of blocksize upon conversion is not yet supported.
                 if b.layout in block_layouts:
                     for block_layout in block_layouts:
-                        with self.assertRaisesRegex(RuntimeError, "blocksize does not match the blocksize"):
+                        with self.assertRaisesRegex(RuntimeError, "conversion from.*to.*is not implemented"):
                             self._convert_to_layout(b, block_layout, blocksize=3)
```

Comment:
Related to my comment above about the early exit and silently incorrect behavior. The `self._convert_to_layout` utility use the `to_sparse_xyz` api directly so it bypassed that early exit and triggers the expected error.

This test should probably be re-written to use `to_sparese(layout, blocksize)` that utility is made obsolete by this PR.

Reply:
Agreed. I have moved the validation checks from `_convert_to_layout` to `test_sparse_to` test method and eliminated `_convert_to_layout` as well as updated other conversion tests.